### DISCUSSION TITLE
Can an Ambianic-edge connect to multiple PWAs simultaneously

### DISCUSSION THREAD

**Discussion by vickywane** [ambianic/ambianic-ui#761] [Q&A] [answered]
I would like to know if the Ambianic-edge can connect to multiple edge PWAs at once. This question is being asked due to an error that is commonly displayed in the browser dev console. 

See error and screenshot attached below;

`PnP service connection error Error: ID "{{REMOTE_ID}}" is taken`
![Screenshot_20211025_201910](https://user-images.githubusercontent.com/29664439/138756646-abb070bd-26a5-4cbe-96ed-cd5a1767780b.png)

This error indicates that the Ambianic Edge cannot be connected to multiple devices at once, but i would like to confirm if this is the design.

**Comment by ivelin** [comment, marked as answer]
> I would like to know if the Ambianic-edge can connect to multiple edge PWAs at once. 

Yes. Multiple tabs in the same browser can open the PWA. Each tab gets its own Peer ID. When the tab is closed, the PWA stops refreshing its lease of the Peer ID with the signaling server (registrar, PnP) and therefore the Peer ID is recycled by the server within a few seconds.

> This question is being asked due to an error that is commonly displayed in the browser dev console.
> 
> See error and screenshot attached below;
> 
> This error indicates that the Ambianic Edge cannot be connected to multiple devices at once, but i would like to confirm if this is the design.

This error indicates that the PWA code within a single browser tab enters an invalid state in which the client is trying to obtain a new Peer ID from the signaling server while it already has one. Look for the condition that leads the client to ask for Peer ID after it is already given one.

**Reply by vickywane** [reply]
Okay. 

I will look for that.

**Comment by ivelin** [comment]
Multiple Edge Device support via PWA is now implemented and merged: 
https://github.com/ambianic/ambianic-ui/pull/757
